vcsutils/webhookparser/bitbucketcloud_test.go

```diff
@@ -37,50 +39,67 @@ func TestBitbucketCloudParseIncomingPushWebhook(t *testing.T) {
-	request.Header.Add(EventHeaderKey, "pullrequest:created")
-
-	// Parse webhook
-	actual, err := ParseIncomingWebhook(vcsutils.BitbucketCloud, token, request)
-	assert.NoError(t, err)
-
-	// Check values
-	assert.Equal(t, bitbucketCloudExpectedPrId, actual.PullRequestId)
-	assert.Equal(t, expectedRepoName, actual.Repository)
-	assert.Equal(t, expectedBranch, actual.Branch)
-	assert.Equal(t, bitbucketCloudPrCreateExpectedTime, actual.Timestamp)
-	assert.Equal(t, expectedRepoName, actual.SourceRepository)
-	assert.Equal(t, expectedSourceBranch, actual.SourceBranch)
-	assert.Equal(t, vcsutils.PrCreated, actual.Event)
-}
-
-func TestBitbucketCloudParseIncomingPrUpdateWebhook(t *testing.T) {
-	reader, err := os.Open(filepath.Join("testdata", "bitbucketcloud", "prupdatepayload"))
-	assert.NoError(t, err)
-	defer closeReader(reader)
-
-	// Create request
-	request := httptest.NewRequest("POST", "https://127.0.0.1?token="+string(token), reader)
-	request.Header.Add(EventHeaderKey, "pullrequest:updated")
-
-	// Parse webhook
-	actual, err := ParseIncomingWebhook(vcsutils.BitbucketCloud, token, request)
-	assert.NoError(t, err)
-
-	// Check values
-	assert.Equal(t, bitbucketCloudExpectedPrId, actual.PullRequestId)
-	assert.Equal(t, expectedRepoName, actual.Repository)
-	assert.Equal(t, expectedBranch, actual.Branch)
-	assert.Equal(t, bitbucketCloudPrUpdateExpectedTime, actual.Timestamp)
-	assert.Equal(t, expectedRepoName, actual.SourceRepository)
-	assert.Equal(t, expectedSourceBranch, actual.SourceBranch)
-	assert.Equal(t, vcsutils.PrEdited, actual.Event)
+func TestBitbucketCloudParseIncomingPrWebhook(t *testing.T) {
+	tests := []struct {
+		name              string
+		payloadFilename   string
+		eventHeader       string
+		expectedTime      int64
+		expectedEventType vcsutils.WebhookEvent
+	}{
+		{
+			name:              "create",
+			payloadFilename:   "prcreatepayload",
+			eventHeader:       "pullrequest:created",
+			expectedTime:      bitbucketCloudPrCreateExpectedTime,
+			expectedEventType: vcsutils.PrCreated,
+		},
+		{
+			name:              "update",
+			payloadFilename:   "prupdatepayload",
+			eventHeader:       "pullrequest:updated",
+			expectedTime:      bitbucketCloudPrUpdateExpectedTime,
+			expectedEventType: vcsutils.PrEdited,
+		},
+		{
+			name:              "merge",
+			payloadFilename:   "prmergepayload",
+			eventHeader:       "pullrequest:fulfilled",
+			expectedTime:      bitbucketCloudPrMergeExpectedTime,
+			expectedEventType: vcsutils.PrMerged,
+		},
+		{
+			name:              "close",
+			payloadFilename:   "prclosepayload",
+			eventHeader:       "pullrequest:rejected",
+			expectedTime:      bitbucketCloudPrCloseExpectedTime,
+			expectedEventType: vcsutils.PrClosed,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			reader, err := os.Open(filepath.Join("testdata", "bitbucketcloud", tt.payloadFilename))
+			assert.NoError(t, err)
+			defer closeReader(reader)
```

Comment:
`closeReader` could be more general if it took a Closer instead of a File. `close(io.Closer)`